### PR TITLE
Various rustfmt and commenting changes

### PR DESCRIPTION
**Pull request by mark-i-m**
These are factored out of #49320

There aren't actually any changes in functionality, just rustfmt and doccomments.

### CODE REVIEW

**Review by estebank — COMMENTED** [review]
LGTM overall, fix the merge conflict and r=me. That being said, I'd lean towards really small rustfmt PRs to minimize the likelihood of causing merge conflicts for open PRs.

File: src/librustc_driver/driver.rs [hunk 1]
```diff
@@ -1299,7 +1373,9 @@ pub fn collect_crate_types(session: &Session, attrs: &[ast::Attribute]) -> Vec<c
     if base.is_empty() {
         base.extend(attr_types);
         if base.is_empty() {
-            base.push(::rustc_trans_utils::link::default_output_for_target(session));
+            base.push(::rustc_trans_utils::link::default_output_for_target(
+                session,
+            ));
```
[the remark this answers is not in the document]
Comment (mark-i-m) [reply src/librustc_driver/driver.rs]: I just ran `rustfmt` on the file. I'm guessing it is avoiding >=100 chars per line...

File: src/librustc_driver/driver.rs [hunk 2]
```diff
@@ -1310,9 +1386,10 @@ pub fn collect_crate_types(session: &Session, attrs: &[ast::Attribute]) -> Vec<c
             let res = !::rustc_trans_utils::link::invalid_output_for_target(session, *crate_type);
 
             if !res {
-                session.warn(&format!("dropping unsupported crate type `{}` for target `{}`",
-                                      *crate_type,
-                                      session.opts.target_triple));
+                session.warn(&format!(
+                    "dropping unsupported crate type `{}` for target `{}`",
+                    *crate_type, session.opts.target_triple
+                ));
```
[the remark this answers is not in the document]
Comment (mark-i-m) [reply src/librustc_driver/driver.rs]: Eh, I could accept either way...

**Comment by mark-i-m** [comment]
@estebank Thanks :) Rebased.

**Comment by mark-i-m** [comment]
@estebank rebased.

**Pull request merged**